### PR TITLE
MAT-3980 truncate over long title and series for test case list display

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-3980](https://jira.cms.gov/browse/MAT-3980)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is into the **correct branch**.
* [x] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [x] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/testCaseList/TestCase.tsx
Comment (adongare) [inline src/components/testCaseList/TestCase.tsx:45]: probably better to create a dumb component that truncates any given text and reuse it wherever needed

Comment (sb-cecilialiu) [reply src/components/testCaseList/TestCase.tsx:45]: @adongare That's a great idea! Thanks a lot I'll work on it!

File: src/components/testCaseList/TestCase.tsx
Comment (adongare) [inline src/components/testCaseList/TestCase.tsx:86]:
```
<td>
   <TruncateText text={testCase.title} maxLength={60} />
</td>
<td>
    <TruncateText text={testCase.series} maxLength={60} />
</td>
```

and then `TruncateText.tsx`

```
import React from "react";
import Button from "@mui/material/Button";
import Tooltip from "@mui/material/Tooltip";
import { truncateInput } from "../../util/Utils";

const TruncateText = ({ text, maxLength = 60 }) => {
  if (text && text.length > maxLength) {
    const displayText = truncateInput(text, maxLength);
    return (
      <Tooltip title={text} placement="right">
        <Button name="title">{displayText}...</Button>
      </Tooltip>
    );
  } else {
    return text;
  }
};

export default TruncateText;

```

Sorry, this increases the work though 😬

**Review by adongare — APPROVED** [review]

File: src/components/testCaseList/TestCase.tsx
Comment (sb-cecilialiu) [reply src/components/testCaseList/TestCase.tsx:86]: @adongare Thanks so much for your help Ashok! I really like the idea! I guess I need to think the React way and this is a great example of it! I appreciate it and will try this component way.

Comment (adongare) [reply src/components/testCaseList/TestCase.tsx:86]: No worries, we all are learning.

Comment (sb-cecilialiu) [reply src/components/testCaseList/TestCase.tsx]: @adongare Had some issues because of the change as well as after the merge, but it's finally done. Please have a look again. I really appreciate your help!

File: src/components/testCaseList/TruncateText.tsx
Comment (adongare) [inline src/components/testCaseList/TruncateText.tsx:10]: both here and down in `else` part, I'm not sure about adding `td` element. This might add `td` inside `td` becauae you have `StyledCell` wrapping this component which already has a `td`. Also, we may not be able to reuse this component anywhere apart from just a table. Rest looks good to me.

**Review by adongare — APPROVED** [review]

File: src/components/testCaseList/TestCase.tsx
Comment (adongare) [reply src/components/testCaseList/TestCase.tsx]: np, thanks!

File: src/components/testCaseList/TruncateText.tsx
Comment (sb-cecilialiu) [reply src/components/testCaseList/TruncateText.tsx:10]: @adongare Thanks Ashok, I didn't realize StyledCell also has a td. I had to merge with develop which had the new td. I'll try remove the td. Thanks!

Comment (adongare) [reply src/components/testCaseList/TruncateText.tsx:10]: np

Comment (sb-cecilialiu) [reply src/components/testCaseList/TruncateText.tsx]: @adongare I removed the td, thanks so much for the suggestion! Please have a look again, appreciate it!!

**Review by adongare — APPROVED** [review]

File: src/components/testCaseList/TruncateText.tsx
Comment (adongare) [reply src/components/testCaseList/TruncateText.tsx]: looks great!

File: src/components/testCaseList/TestCase.tsx
Comment (sb-cecilialiu) [reply src/components/testCaseList/TestCase.tsx]: @adongare Hi Ashok! I see you approve the pull request, is it ok if I click the Resolve conversion and merge the PR? Thanks so much for your constant help!!

Comment (adongare) [reply src/components/testCaseList/TestCase.tsx]: Sorry, yes. forgot to resolve this.

**Pull request merged**
